### PR TITLE
[JENKINS-52024] - Update Metainf services from 1.4 to 1.8 in the master branch

### PR DESCRIPTION
**Pull request by oleg-nenashev**
See [JENKINS-52024](https://issues.jenkins-ci.org/browse/JENKINS-52024) submitted to Java 10+ support. Originally it was submitted by @svanoort in https://github.com/jenkinsci/jenkins/pull/3516 to fix Jenkins core compilation.

### Proposed changelog entries

* RFE, Internal: Update Metainf services from 1.4 to 1.8 to fix compilation on JDK 10+
  * Component link: https://github.com/kohsuke/metainf-services

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@svanoort @jglick @dwnusbaum

### CODE REVIEW

**Review by dwnusbaum — APPROVED** [review]

**Review by svanoort — APPROVED** [review]
Go for it!

File: cli/pom.xml [hunk 1]
```diff
@@ -33,6 +33,11 @@
         <groupId>org.kohsuke</groupId>
         <artifactId>access-modifier-annotation</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci</groupId>
+      <artifactId>annotation-indexer</artifactId>
+      <version>1.12</version>
```
Comment (jglick) [inline cli/pom.xml]: Please either use `dependencyManagement`, or remove this dep from `jenkins-core` and have it inherit from `cli`.

**Review by jglick — DISMISSED** [review]

File: cli/pom.xml (hunk 1 above)
Comment (dwnusbaum) [reply cli/pom.xml]: The explicit dependency would be unnecessary if kohsuke/access-modifier#15 was merged and released and we picked up that version of access-modifier, but `dependencyManagement` seems like a better approach anyway, so +1.

**Comment by oleg-nenashev** [comment]
@jglick WDYT?

**Review by svanoort — APPROVED** [review]

**Pull request merged**